configure.ac

```diff
@@ -39,6 +39,10 @@ AC_CONFIG_HEADERS([src/config.h])
 AC_CANONICAL_HOST
 
 OPAL_CHECK_ATTRIBUTES
+AS_IF([test "$opal_cv___attribute__deprecated" = 1],
+      [SHMEM_HAVE_ATTRIBUTE_DEPRECATED=1],
+      [SHMEM_HAVE_ATTRIBUTE_DEPRECATED=0])
+AC_SUBST(SHMEM_HAVE_ATTRIBUTE_DEPRECATED)
```

Comment:
What about the naming of the autoconf substitution variables?  We have ```SHMEM_HAVE_ATTRIBUTE_DEPRECATED``` and ```SHMEM_C_HAVE_ATTRIBUTE_VISIBILITY```.  Might not be a bad idea to distinguish these as autoconf output so there isn't a name clash.  How about ```SHMEM_AC_HAVE_...```?